src/types/select.ts

```diff
@@ -0,0 +1,25 @@
+import { AnyElementOf } from '@/types/utils'
+
+export interface SelectItem {
```

Comment:
question: Can this interface extend `Record<string, any>`? Either that or can we have a `record` prop (optional) that can contain the full object data (we would also want this for MultiselectItem). Sometimes there is a need to reference a property that is being used for the `label` or the `value`.